File: ALODWebUtility/Common/PSIDUtility.vb
Imports ALOD.Core.Domain.Common
Imports ALOD.Core.Domain.Modules.SpecialCases
Imports ALOD.Data.Services

Namespace Common
    Public Module PSIDUtility
        Public Function HasFinding(ByVal finding As SC_PSCD_Findings) As Boolean
            If (finding Is Nothing) Then
                Return False
            End If

            If (Not finding.Finding.HasValue OrElse finding.Finding.Value = 0) Then
                Return False
            End If

            Return True
        End Function

        Public Function IsTimeToSaveViaNavigator(ByVal buttonType As NavigatorButtonType) As Boolean
            If (buttonType = NavigatorButtonType.Save OrElse
                buttonType = NavigatorButtonType.NavigatedAway OrElse
                buttonType = NavigatorButtonType.NextStep OrElse
                buttonType = NavigatorButtonType.PreviousStep) Then
                Return True
            End If

            Return False
        End Function

        Public Function CreatePSIDFinding(ByVal psidID As Integer) As SC_PSCD_Findings
            Dim cFinding As SC_PSCD_Findings = New SC_PSCD_Findings()
            Dim currUser = UserService.CurrentUser

            cFinding.PSIDId = psidID
            cFinding.Name = currUser.FullName
            cFinding.ModifiedBy = currUser.Id
            cFinding.ModifiedDate = DateTime.Now
            cFinding.CreatedBy = currUser.Id
            cFinding.CreatedDate = DateTime.Now

            Return cFinding
        End Function
    End Module
End Namespace

File: ALODWebUtility.CS/Common/PSIDUtility.cs
using System;
using ALOD.Core.Domain.Common;
using ALOD.Core.Domain.Modules.SpecialCases;
using ALOD.Data.Services;

namespace ALODWebUtility.Common
{
    public static class PSIDUtility
    {
        public static bool HasFinding(SC_PSCD_Findings finding)
        {
            if (finding == null)
            {
                return false;
            }

            if (!finding.Finding.HasValue || finding.Finding.Value == 0)
            {
                return false;
            }

            return true;
        }

        public static bool IsTimeToSaveViaNavigator(NavigatorButtonType buttonType)
        {
            if (buttonType == NavigatorButtonType.Save ||
                buttonType == NavigatorButtonType.NavigatedAway ||
                buttonType == NavigatorButtonType.NextStep ||
                buttonType == NavigatorButtonType.PreviousStep)
            {
                return true;
            }

            return false;
        }

        public static SC_PSCD_Findings CreatePSIDFinding(int psidID)
        {
            SC_PSCD_Findings cFinding = new SC_PSCD_Findings();
            var currUser = UserService.CurrentUser();

            cFinding.PSCDId = psidID;
            cFinding.Name = currUser.FullName;
            cFinding.ModifiedBy = currUser.Id;
            cFinding.ModifiedDate = DateTime.Now;
            cFinding.CreatedBy = currUser.Id;
            cFinding.CreatedDate = DateTime.Now;

            return cFinding;
        }
    }
}
